### PR TITLE
[QOLDEV-554] fix overly aggressive timestamp parsing

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Use our guessed types to restrict the behaviour of the numeric/timestamp converter, so columns that sniff as text won't be incorrectly partially converted to timestamps.

We still want to use our aggressive parser when sniffing types, so we convert whatever we can and see what we get. But once sniffing is done and we're doing the actual loading, use a less aggressive post-processor that remembers what types we used and only converts when wanted.

### CODE REVIEW

**Comment by duttonw** [comment]
any extra tests to 'test' the hh:mm not being converted if set to 'text' ?

**Comment by ThrawnCA** [comment]
> any extra tests to 'test' the hh:mm not being converted if set to 'text' ?

There is a test now.

**Review by duttonw — APPROVED** [review]

File: ckanext/xloader/parser.py [hunk 1]
```diff
@@ -1,161 +1,71 @@
...
 from decimal import Decimal, InvalidOperation
-from itertools import chain
+import re
+import six
 
 from ckan.plugins.toolkit import asbool
-from dateutil.parser import isoparser, parser
-from dateutil.parser import ParserError
-
-from tabulator import helpers
```
Comment (duttonw) [inline ckanext/xloader/parser.py:10]: what else was in the parser we don't have now.

Comment (ThrawnCA) [reply ckanext/xloader/parser.py:10]: I did a diff between our parser and the vanilla one. Adding the conversion to numbers and dates was it.

**Pull request merged**